wandb/sdk/artifacts/artifact.py

```diff
@@ -884,36 +896,57 @@ def _update(self) -> None:
                 for alias in self._aliases
             ]
 
-        mutation = gql(
-            """
-            mutation updateArtifact(
-                $artifactID: ID!,
-                $description: String,
-                $metadata: JSONString,
-                $aliases: [ArtifactAliasInput!]
+        mutation_template = """
+        mutation updateArtifact(
+            $artifactID: ID!,
+            $description: String,
+            $metadata: JSONString,
+            _TTL_DURATION_SECONDS_TYPE_
+            $aliases: [ArtifactAliasInput!]
+        ) {
+            updateArtifact(
+                input: {
+                    artifactID: $artifactID,
+                    description: $description,
+                    metadata: $metadata,
+                    _TTL_DURATION_SECONDS_VALUE_
+                    aliases: $aliases
+                }
             ) {
-                updateArtifact(
-                    input: {
-                        artifactID: $artifactID,
-                        description: $description,
-                        metadata: $metadata,
-                        aliases: $aliases
-                    }
-                ) {
-                    artifact {
-                        id
-                    }
+                artifact {
+                    id
                 }
             }
-            """
-        )
+        }
+        """
+        api = InternalApi()
+        fields = api.server_create_artifact_introspection()
+        if "ttlDurationSeconds" in fields:
+            mutation_template = mutation_template.replace(
+                "_TTL_DURATION_SECONDS_TYPE_", "$ttlDurationSeconds: Int64,"
+            ).replace(
+                "_TTL_DURATION_SECONDS_VALUE_",
+                "ttlDurationSeconds: $ttlDurationSeconds,",
+            )
+        else:
+            logger.error(
```

Comment:
Nit: this should be a warning not an error, since it doesn't stop the artifact creation. Also, let's only show it if someone actually tries to set a TTL.